lib/gcloud/datastore/entity.rb

```diff
@@ -216,9 +222,9 @@ def map_exclude_flag_to_value flag, value #:nodoc:
 
       ##
       # Update the exclude data after a new object is created.
-      def update_exclude_indexes! #:nodoc:
+      def update_exclude_indexes! entity #:nodoc:
         @_exclude_indexes = {}
-        Array(@_entity.property).each do |property|
+        Array(entity.property).each do |property|
           @_exclude_indexes[property.name] = property.value.indexed
           unless property.value.list_value.nil?
             exclude = Array(property.value.list_value).map(&:indexed)
```

Comment:
Yep. We support that now actually. Take a look at [test/gcloud/datastore/test_entity_exclude.rb](https://github.com/GoogleCloudPlatform/gcloud-ruby/blob/master/test/gcloud/datastore/test_entity_exclude.rb) for the covered cases. You can set one exclude value to be used for each value the list:

```ruby
entity["tags"] = ["ruby", "code"]
entity.exclude_from_indexes! "tags", true
entity.exclude_from_indexes?("tags") #=> [true, true]
```

Or you can set a list of exclude values:

```ruby
entity["tags"] = ["ruby", "code"]
entity.exclude_from_indexes! "tags", [true, false]
entity.exclude_from_indexes?("tags") #=> [true, false]
```

Or, my favorite, you can pass a  block with the exclude logic to be evaluated on each value in the list:

```ruby
entity["tags"] = ["ruby", "code"]
entity.exclude_from_indexes! "tags" do |tag|
  tag =~ /r/
end
entity.exclude_from_indexes?("tags") #=> [true, false]
```